py/bitbox02/bitbox02/communication/bitbox_api_protocol.py

```diff
@@ -214,6 +245,9 @@ def __init__(
                     self.close()
                     raise Exception("Unlock process aborted")
 
+        # raises exceptions if either firmware or library are out of date, does not check BitBoxBase
+        self.check_version()
```

Comment:
This should be moved to above the unlock and attestation check, as you might never come this far if the firmware version is too new and those two calls change (which they will in 7.0.0 :laughing: )